File: 3834-minimum-operations-to-convert-all-elements-to-zero/3834-minimum-operations-to-convert-all-elements-to-zero.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int res = 0;
        stack<int> stk;
        for(int i = 0; i < nums.size(); i++)
        {
            while(!stk.empty() && nums[i] < stk.top())
            {
                stk.pop();
            }
            if((stk.empty() || nums[i] > stk.top()) && nums[i] > 0)
            {
                stk.push(nums[i]);
                res++;
            }
        }
        return res;
    }
};

File: 3834-minimum-operations-to-convert-all-elements-to-zero/3834-minimum-operations-to-convert-all-elements-to-zero.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        res = 0
        stk = []
        for num in nums:
            while len(stk) > 0 and num < stk[-1]:
                stk.pop()
            if (len(stk) == 0 or num > stk[-1]) and num > 0:
                stk.append(num)
                res += 1
        return res

                    